saplings/circuits/src/components/ServiceCard.js

```diff
@@ -0,0 +1,576 @@
+/**
+ * Copyright 2018-2020 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useReducer, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Service } from '../data/circuits';
+import { Node } from '../data/nodeRegistry';
+import { PlusButton, MinusButton } from './PlusMinusButton';
+
+import './ServiceCard.scss';
+
+const isValidID = serviceID => {
+  if (serviceID.length === 4) {
+    const regex = /^[a-z0-9]+$/i;
```

Comment:
Aren't capital letters valid, as well?

Reply:
Fixed. I changed the generateServiceID function as well, as before it was only generating IDs with lower case letters